pkg/controller/deployment/deployment_controller.go

```diff
@@ -438,41 +438,124 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	if d.Spec.Paused {
-		// TODO: Implement scaling for paused deployments.
-		// Don't take any action for paused deployment.
-		// But keep the status up-to-date.
-		// Ignore paused deployments
-		glog.V(4).Infof("Updating status only for paused deployment %s/%s", d.Namespace, d.Name)
-		return dc.syncPausedDeploymentStatus(d)
+		return dc.sync(d)
 	}
+
 	if d.Spec.RollbackTo != nil {
 		revision := d.Spec.RollbackTo.Revision
 		if _, err = dc.rollback(d, &revision); err != nil {
 			return err
 		}
 	}
 
+	if dc.isScalingEvent(d) {
+		return dc.sync(d)
```

Comment:
The only difference between the rollout code and `scale` (apart from the way we scale) is that in `scale` we will not create a new replica set (ie. we will not react on pod template changes). This should always be the default behavior when the deployment is paused. When the deployment is unpaused then we will not enter [scale](https://github.com/kubernetes/kubernetes/pull/20273/commits/47ee2c8c64b0d7869d4a7c668313f9e2b46b3d7c) (a missing new rs means there is a change in the pod template of the deployment and we need to rollout.